### DISCUSSION TITLE
Are there way to connect multiple GoPros simultaneously??

### DISCUSSION THREAD

**Discussion by sunwonlikeyou** [gopro/OpenGoPro#71] [Q&A] [answered]
I'd like to connect multiple gopros simultaneously.
That means I can get live-streams with same time-stamps from multiple gopros.
and I wonder connection speed in your settings. Max connection speed or max frame rate with multiple gopros.

Thank you.

**Comment by tcamise-gpsw** [comment]
Hello. Is this in reference to a specific implementation or demo? 

If not, the Open GoPro API does specify any limitations on the peer device (i.e. the device that connects to the GoPros) so this is entirely dependent on your implementation. At least there is nothing in the API that prevents connecting to simultaneous GoPros

**Reply by KonradIT** [reply]
I've tried using the bleak library to control multiple devices in a queue, so not asynchronously. https://github.com/KonradIT/gopro-ble-py/tree/v2#multi-camera

**Reply by patrick91** [reply]
@tcamise-gpsw where can I find the limitations? I've tried to find them in the docs, but I'm unable to find anything related to that.

Ideally I'd like to preview videos from multiple gopro at once, I don't really mind what language I need to use, but I'm unclear whether this is possible via BLE or if I need multiple wifi cards to connect to each go pro

**Comment by tcamise-gpsw** [comment, marked as answer]
From the GoPro's perspective, each GoPro will only be connected to one device so this is no different than the normal use case.
The limitations that you need to consider are from the central device that is managing the connections to all of the GoPro's.
If you are using a computer, I can't imagine any problems for WiFi. And likely no problems for BLE at least until you hit dozens of connections.
If you are using, for example, an embedded device then you are constrained by its resources.

In short, there is nothing special needed from the GoPro side here.

**Comment by jonericcook** [comment]
Hi @tcamise-gpsw 👋 

I wanted to ask a follow up question as it pertains to my use case because the above conversation and the two items below have me a bit confused:

1. `The socket address for WiFi connections is 10.5.5.9:8080` - https://gopro.github.io/OpenGoPro/http_2_0#wifi-2
2. Question: `Is there currently a way to connect multiple cameras on the same Wifi network?` -- Answer: `No. Cameras can only be connected through Wi-Fi by becoming an access point itself (generating its own Wi-Fi network), not as a peripheral.` - https://gopro.github.io/OpenGoPro/faq#multi-camera-setups

For point number 1 does that mean every gopro that connects to wifi has the same IP address?

For point number 2 does that mean I can't have multiple gopros join the same wifi network?

-------

My use case / setup:
- computer
- 5Ghz wifi router that will only have my pc (connected to the back of the router via ethernet) and 4 gopros connected to it
- write a python script that will:
  -  call out to each camera (over wifi) to start recording a video
  - after N numbers of seconds call out to each camera (over wifi) to stop recording
  - call out to each camera (over wifi) to download the video that was just recorded to the computer

As a side note using USB-C doesn't appear to be an option as the gopros will be 20 ft away from the computer.

------

Thank you guys for making OpenGoPro - it's such an awesome tool 💯

**Reply by deenisbb** [reply]
Did you manage to solve your problem?

**Reply by jonericcook** [reply]
Nope - i opted to use a different camera. I didn't want to move forward with GoPro without my questions being answered.
